src/game.cpp

```diff
@@ -14561,11 +14561,11 @@ void game::start_calendar()
         if( scen->has_flag("SPR_START") ) {
             ; // Do nothing;
         } else if( scen->has_flag("SUM_START") ) {
-            calendar::start += DAYS((int)ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]);
+            calendar::start += DAYS( calendar::season_length() );
```

Comment:
> whilst  calendar::set_season_length  is invoked from fairly weird places ( options_manager::serialize  and  worldfactory::set_active_world ).

It gets executed when the contents of `ACTIVE_WORLD_OPTIONS` changes. Those are exactly the places where it must be.

> It should probably be placed here and be initialized from  ACTIVE_WORLD_OPTIONS  too.

It needs to be set when the player changes the world options while already playing the game, therefor in `options_manager::serialize` (called from `options_manager::show`).

Through it should probably be set directly in `options_manager::show` - but that makes not actual difference for this PR.

> These changes (x6) don't look consistent. The function's name is  start_calendar()  so it's supposed to initialize it, not rely on its values.

The function "starts" `calendar::start` and `calendar::turn` (actual `calendar` objects). `calendar::season_length` is not really part of them, it's a static function, accessing global data.

For what it's worth, `calendar::season_length` could as well be a global function, or a game function or whatever. 

Actually `calendar::turn` and `calendar::start` should probably be moved to `game`.

----

`calendar::season_length` is a simple wrapper around `ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]`, nothing more. The PR is supposed to replace the separate ways of accessing the option value (via `ACTIVE_WORLD_OPTIONS` and via `season_length`) with a single (established) way.

You can see it like that: where do you look if you want to know the season length? The options or the calendar? How do you even know it's an option?